doc/extending/extending_theano.txt

```diff
@@ -876,11 +876,24 @@ Modify and execute the example to return two outputs: x + y
     and x - y.
 
 
-Documentation
--------------
+.. _Documentation:
 
-See :ref:`metadocumentation`, for some information on how to generate
-the documentation.
+Documentation and Coding Style
+------------------------------
+
+* The code should be compatible with Python 2.6 and above, as well as Python 
+  3.3 and above (using `six` if needed).
+
+* All the code should respect the `PEP8 Code Style Guide <http://www.python.org/dev/peps/pep-0008>`_.
+
+* The docstrings of all the classes and functions should respect the
+  `PEP257 <https://www.python.org/dev/peps/pep-0257/>`_ rules and follow the `Numpy docstring standard
+  <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_.
+
+* To cross-reference other objects (e.g. reference other classes or methods) in the docstrings, use the
+  `cross-referencing objects <http://www.sphinx-doc.org/en/stable/domains.html#cross-referencing-python-objects>`_ syntax. ``:py`` can be omitted, see e.g. this `stackoverflow answer <http://stackoverflow.com/a/7754189>`_.
+
+* See :ref:`metadocumentation`, for some information on how to generate the documentation.
 
```

Comment:
I moved things around as we discussed IRL yesterday. Everything is in the *dev_start_guide* now and the *extending_theano* section refers to it. I also changed the order of the information to prevent people from skipping it.